File: my-submissions/m2285 v2.py
class Solution:
    def maximumImportance(self, n: int, roads: List[List[int]]) -> int:
        cnt = [0] * n
        for road in roads :
            cnt[road[0]] += 1
            cnt[road[1]] += 1
        cnt.sort()

        i = n
        output = 0

        while cnt and cnt[-1] > 0 :
            output += i * cnt.pop()
            i -= 1

        return output

File: my-submissions/m2285 v2.java
class Solution {
    public long maximumImportance(int n, int[][] roads) {
        int[] cnt = new int[n];

        for (int[] road : roads) {
            cnt[road[0]]++;
            cnt[road[1]]++;
        }
        Arrays.sort(cnt);

        long output = 0;
        for (int i = n; i > 0 && cnt[i - 1] > 0; i--) {
            output += ((long) i * (long) cnt[i - 1]);
        }
        return output;
    }
}
